src/pages/index.js

```diff
@@ -65,34 +109,137 @@ profileEditBtn.addEventListener("click", () => {
+);
+
+profilePictureModal.setEventListeners();
 
-const addModal = new PopupWithForm("#profile-add-modal", () => {
-  const { title, url } = addModal.getInputValues();
-  const newCard = renderCard({ name: title, link: url });
-  cardSection.addItem(newCard);
-  addModal.close();
+profilePictureEditBtn.addEventListener("click", () => {
+  profilePictureModal.open();
 });
 
+//--------------------ADD CARD POPUP-WITH-FORM INSTANTIATION-------------------->>
+
+const addModal = new PopupWithForm(
+  "#profile-add-modal",
+  () => {
+    addModal.loadingButtonState();
+    const { title, url } = addModal.getInputValues();
+    api
+      .addNewCard({ name: title, link: url })
+      .then((cardData) => {
+        const newCard = renderCard({
+          name: cardData.name,
+          link: cardData.link,
+          _id: cardData._id,
+          isLiked: cardData.isLiked,
+        });
+        cardSection.addItem(newCard);
+        addModal.resetButtonState();
+      })
+      .catch((err) => {
+        api.logError(err);
+      });
+    addModal.close();
+  },
+  "Create"
+);
+
 addModal.setEventListeners();
 
 addCardBtn.addEventListener("click", () => {
   addModal.open();
 });
 
+//--------------------DELETE CARD POPUP INSTANTIATION-------------------->>
+
+const deleteModal = new PopupWithConfirmation("#card-delete-modal");
+
+deleteModal.setEventListeners();
+
 //--------------------FUNCTIONS-------------------->>
 
 function renderCard(data) {
-  const cardElement = new Card(data, "#card-template", () => {
-    imagePopup.open(data);
-  });
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    () => {
+      imagePopup.open(data);
+    },
+    deleteModal,
+    () => {
+      const id = cardElement.getId();
+
+      deleteModal.open();
+
+      deleteModal.setSubmitAction(() => {
+        deleteModal.loadingButtonState();
+
+        api
+          .deleteCard(id)
+          .then(() => {
+            cardElement.deleteHandler();
+            deleteModal.close();
+            deleteModal.resetButtonState();
+          })
+          .catch((err) => {
+            api.logError(err);
+          });
+      });
+    },
+    () => {
+      const id = cardElement.getId();
+
+      if (!data.isLiked) {
+        api
+          .addCardLike(id)
+          .then(() => {
+            cardElement.likeHandler();
+          })
+          .catch((err) => {
+            api.logError(err);
+          });
+      } else {
+        api
+          .removeCardLike(id)
+          .then(() => {
+            cardElement.likeHandler();
+          })
+          .catch((err) => {
+            api.logError(err);
+          });
+      }
+    }
+  );
   return cardElement.getNewCard();
 }
 
 //--------------------VALIDATION INSTANTIATORS-------------------->>
 
 const profileEditValidation = new FormValidator(options, profileEditForm);
 const addCardValidation = new FormValidator(options, cardAddForm);
+const profilePictureValidation = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```